### DISCUSSION TITLE
Filters - add "Empty" checkbox option to the filter

### DISCUSSION THREAD

**Discussion by vkuybida** [microsoft-search/pnp-modern-search#3669] [Q&A]
We have a filter based on Taxonomy field values.  We have request to add "Empty" filter option checkbox (so once refined, search will display all the items with empty value for that Taxonomy field). What would be the best option to implement it in Filters web part?

![image](https://github.com/microsoft-search/pnp-modern-search/assets/12577514/926cd8f7-f944-4302-a8b5-e323ee55d233)

**Comment by kasperbolarsen** [comment]
HI, the short version: You can't!!

The slightly longer version: As far as I know your best option would be to add a search vertical for each of the refiners where you want to find the content that does not have a value in that column.
Let's say that your current query is something similar to this: 
{SearchTerms} contenttypeid:0x3928093* Path:https://contoso.sharepoint.com/sites/PnPModernSearch/SearchableDocuments*
This will give you a number of hits
The trick is to modify the query to show every result that has a value from the term ser in question ( like the restaurent example in this, https://www.techmikael.com/2014/03/s15e01-kql-basics.html)
You have to figure out which managed property that represents the term set. In my case below the internal name of the site column is searchDepartment, and the managed property is owstaxIdsearchDepartment.

Once that is handled I can test my query:

{searchTerms} ContentTypeID:0x01010093DC4846A20EA24293E2242142D86BD8* path:https://contoso.sharepoint.com/sites/PnPModernSearch/SearchableDocuments*
owstaxIdsearchDepartment:"#8ed8c9ea-7052-4c1d-a4d7-b9c10bffea6f" 

This will give me all the documents that has the content type, is in the specified library and has a value from the Department term set
![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/d3c5d551-3060-4788-bcec-12de3f33ea3f)

The final step is to invert the last segment, as I want the documents that does NOT have a value from the term set:
{searchTerms} ContentTypeID:0x01010093DC4846A20EA24293E2242142D86BD8* path:https://contoso.sharepoint.com/sites/PnPModernSearch/SearchableDocuments*
-owstaxIdsearchDepartment:"#8ed8c9ea-7052-4c1d-a4d7-b9c10bffea6f" 

Voila, and I hopes this makes sense :-)

**Reply by vkuybida** [reply]
Thanks for the answer. Yes, I understand that we can filter data using search query changes. But we have a request to have it as a filter option. Is it possible to achieve it using an extensibility approach (e.g. build a custom UI component?)

**Reply by wobba** [reply]
May be possible with a custom refiner template? Where you set the value to be blank.
